### PR TITLE
[dotnet-embedding] add documentation re why bitcode is not supported

### PR DESCRIPTION
**Pull request by whitneyschmidt**

### CODE REVIEW

**Comment by PRMerger14** [comment]
@whitneyschmidt : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Review by spouliot — CHANGES_REQUESTED** [review]
That's only true for iOS applications (not for Xamarin.iOS).
E.g. for tvOS and watchOS we generate bitcode because it's a requirement to submit applications to the app store.

**Comment by PRMerger10** [comment]
@whitneyschmidt : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/tools/dotnet-embedding/limitations.md [hunk 1]
```diff
@@ -44,4 +44,10 @@ Since we cannot generate accurate nullability annotations in the header files an
 
 Currently .NET Embedding does not support bitcode on iOS, which is enabled for some Xcode project templates. This will have to be disabled to successfully link generated frameworks.
 
+There are a couple reasons why .NET Embedding for iOS is not supported:
```
Comment (spouliot) [inline docs/tools/dotnet-embedding/limitations.md]: >  why .NET Embedding for iOS is not supported:

No, embedding is supported! It's **bitcode for iOS** that is not supported.

File: docs/tools/dotnet-embedding/limitations.md [hunk 2]
```diff
@@ -44,4 +44,10 @@ Since we cannot generate accurate nullability annotations in the header files an
...
+
+* Xamarin can generate bitcode for iOS, but it is inline assembly, not "real" bitcode. Xcode can recompile iOS bitcode, but not optimize or modify it.
```
Comment (spouliot) [inline docs/tools/dotnet-embedding/limitations.md]: >  Xamarin can generate bitcode for iOS

No, we could but we don't - for the reason I mentioned on slack.

**Review by spouliot — CHANGES_REQUESTED** [review]
I think you can (mostly) copy/paste my text from slack. Here's a more complete / updated version:

> For iOS bitcode is optional to submit Apps to Apple's AppStore. Xamarin.iOS does not support it for iOS (only) since the generated bitcode is “inline assembly”. This provides no benefit on the iOS platform, i.e. it can’t be optimized (server-side), but makes binaries larger and build times longer. 

> For tvOS and watchOS bitcode is required for submitting Apps to Apple's AppStore. Xamarin.iOS supports bitcode on tvOS (as "inline assembly") and watchOS (as "LLVM/IR") to fulfill this requirement.

> For macOS bitcode support is currently not required, nor supported by Xamarin.Mac.

File: docs/tools/dotnet-embedding/limitations.md (hunk 2 above)
Comment (spouliot) [reply docs/tools/dotnet-embedding/limitations.md]: Saying "can" implies the tool can do it - which is what the customer are asking, without understanding that
*  they would not get he theoretical benefits (server side optimization);
* they would suffer longer build times (to generated bitcode) and binary sizes (which now include bitcode, before Apple re-process them)

**Comment by PRMerger12** [comment]
@whitneyschmidt : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Review by spouliot — APPROVED** [review]

File: docs/tools/dotnet-embedding/limitations.md [hunk 3]
```diff
@@ -44,4 +44,8 @@ Since we cannot generate accurate nullability annotations in the header files an
 
 Currently .NET Embedding does not support bitcode on iOS, which is enabled for some Xcode project templates. This will have to be disabled to successfully link generated frameworks.
 
+For iOS bitcode is optional to submit Apps to Apple's AppStore. Xamarin.iOS does not support it for iOS (only) since the generated bitcode is “inline assembly”. This provides no benefit on the iOS platform, i.e. it can’t be optimized (server-side), but makes binaries larger and build times longer.
```
Comment (davidortinau) [inline docs/tools/dotnet-embedding/limitations.md]: should "Apps" be uppercase? I think "apps".

> (only)
seems confusing. I would just remove it or clarify it further 

For all these, should they perhaps be bullets? When previewed it's all one big paragraph, and I liked them better separately. Easier to scan (which is how docs are typically read, right?).

The initial dependent clauses need a comma on each, like... "For iOS, bitcode" and "For tvOS and watchOS, bitcode"
server-side), but makes binaries larger and build times longer.

**Review by davidortinau — APPROVED** [review]

File: docs/tools/dotnet-embedding/limitations.md (hunk 3 above)
Comment (spouliot) [reply docs/tools/dotnet-embedding/limitations.md]: Don't trust anything I write in English, or in French for that matter ;-)
Even less for formatting... I would have more success using a RNG to place them.

I remember some conversation about "Apps" with Vincent and I think it's not very consistent across Apple. Funnily that's something easier in French - which is quite uncommon :)

**Comment by PRMerger18** [comment]
@whitneyschmidt : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by conceptdev** [comment]
#sign-off (per @davidortinau above)

**Pull request merged**